### PR TITLE
Trm 2020 0621

### PR DESCRIPTION
**Pull request by TimMangan**
Two changes included:

1. Allow debug builds of PsfLauncher to accept a new config.json entry for WaitForDebugger. This simplifies the task of debugging launch issues with the launcher by waiting as soon as possible.
2. Fix remaining PrivateProfile functions that crash the app when it calls with a NULL filename argument, or possibly other NULL arguments.

### CODE REVIEW

**Comment by dhoehna** [comment]
Hey Tim.  This PR has the changes from your last PR in it as well.  For example your PR had all the ECHO statements in buildall.cmd.

Can you please clean up the PR so only new changes are going into develop?

Thanks.

**Comment by TimMangan** [comment]
This PR was created out of a fresh branch of the Microsoft develop branch on Sunday, with only the new files above changed after; PsfLauncher (main.cpp and Readme.md in 16f169b) and FileRedirectionFixup (*PrivateProfile* from 3e6fd60.  740de11, 6562e52, and e66ac6c are a wash, as the first added two new files into the wrong folder and then I removed them.  So I'm not sure why you see any other changes.

**Comment by dhoehna** [comment]
I see why all the changes from your PR are in this one.  Take a look at your commit history.  This branch has the PR from develop to master.

May I ask how you made a fresh copy?  My hunch of why the develop->master PR is included is because your local copy of develop wasn't updated after I accepted your PR.

**Comment by TimMangan** [comment]
Yes. I created a new branch for #146 and did the PR from there, and then made a new branch from Microsoft\develop and made the changes for this one.  I wasn't aware that my develop branch would be in play when I make a PR from Trm_2020_0621 into Microsoft/Develop.  I'll try to figure out how to update my develop branch.  To be clear, do I want it to look like Microsoft/Develop or TRM-2020_0621?

**Comment by dhoehna** [comment]
Hey Tim,

Thank you for the reply.  I see the problem.  When I accept a PR into develop what your computer sees as microsoft/develop is the state before the PR.  You would need to get those changes into your computer.  You can do this with either a git fetch or git pull.  I'd prefer git fetch since that only pulls the references down and thus, will give you the updated develop.

What you'll need to do is fetch the changes from microsoft/develop. Here is how to do that.

1. git remote add upstream https://github.com/microsoft/MSIX-PackageSupportFramework.git (You may have done this already).  This step makes a link to microsoft/develop and gives it a name of upstream.
2. git fetch upstream.  This step will take the history of microsoft/develop and store it on your computer.  

From here you can rebase microsoft/develop onto your develop branch.  If you are going to rebase onto the branch that you are using for the PR you'll need to do a force push to push the changes to your repo.
3. git push -f origin trm-2020-0621

Basically, the gist is that your references to microsoft/develop does not automatically update when a PR is accepted and you'll need to fetch those changes manually.

Hope this helps.

**Comment by TimMangan** [comment]
OK.  I'm running a training class so it might take a couple of days for me to sort this out.

By the way, I did notice that some/all??? of PR146 seems to be in the Microsoft/Master branch. Did you migrate to there after processing 146 to Microsoft/Develop, or possibly is the issue actually in how PR146 was processed?

**Comment by dhoehna** [comment]
Hey Tim,

You observed correctly.  I'm moving away from batch updates to master because a batch update removes the history on who made what change.

Develop was supposed to be the gate, to make sure that all changed going into master are tested and run correctly.  But, it is an extra step for me and the PR's look weird.  

In the future I plan to remove develop, but not until this repo is up to my quality of standard.

**Comment by TimMangan** [comment]
Replacing this PR with PR149 due to fetch issues.
